File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/Relationship.java
package work.socialhub.kmastodon.entity;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * @author hecateball
 */
public class Relationship implements Serializable {

    @SerializedName("id")
    private String id;
    @SerializedName("following")
    private boolean following;
    @SerializedName("followedBy")
    private boolean followedBy;
    @SerializedName("blocking")
    private boolean blocking;
    @SerializedName("muting")
    private boolean muting;
    @SerializedName("requested")
    private boolean requested;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public boolean isFollowing() {
        return following;
    }

    public void setFollowing(boolean following) {
        this.following = following;
    }

    public boolean isFollowedBy() {
        return followedBy;
    }

    public void setFollowedBy(boolean followedBy) {
        this.followedBy = followedBy;
    }

    public boolean isBlocking() {
        return blocking;
    }

    public void setBlocking(boolean blocking) {
        this.blocking = blocking;
    }

    public boolean isMuting() {
        return muting;
    }

    public void setMuting(boolean muting) {
        this.muting = muting;
    }

    public boolean isRequested() {
        return requested;
    }

    public void setRequested(boolean requested) {
        this.requested = requested;
    }

}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/Relationship.kt
package work.socialhub.kmastodon.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
class Relationship {

    @SerialName("id")
    var id: String? = null

    @SerialName("following")
    var isFollowing: Boolean = false

    @SerialName("followedBy")
    var isFollowedBy: Boolean = false

    @SerialName("blocking")
    var isBlocking: Boolean = false

    @SerialName("muting")
    var isMuting: Boolean = false

    @SerialName("requested")
    var isRequested: Boolean = false
}
